src/org/ohdsi/usagi/ui/ShowReviewStatsDialog.java

```diff
@@ -0,0 +1,165 @@
+ ******************************************************************************/
+package org.ohdsi.usagi.ui;
+
+import org.ohdsi.usagi.CodeMapping;
+import org.ohdsi.utilities.collections.Pair;
+
+import javax.swing.*;
+import java.awt.*;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShowReviewStatsDialog extends JDialog {
+
+	private static final long	serialVersionUID	= 2028328868610404663L;
+
+	public ShowReviewStatsDialog() {
+		Font headerFont = new Font("Arial", Font.BOLD,12);
+
+		java.util.List<CodeMapping> selectedCodeMappings = Global.mappingTablePanel.getSelectedCodeMappings();
+		List<CodeMapping> codeMappings;
+		if (selectedCodeMappings.size() > 1) {
+			codeMappings = selectedCodeMappings;
+		} else {
+			codeMappings = Global.mapping;
+		}
+
+		setTitle("Review statistics");
+		setLayout(new GridBagLayout());
+
+		GridBagConstraints g = new GridBagConstraints();
+		g.fill = GridBagConstraints.BOTH;
+		g.ipadx = 50;
+		g.ipady = 10;
+
+		g.gridx = 0;
+		g.gridy = 0;
+		add(new JLabel(String.format("Number of (selected) source codes: %d", codeMappings.size())), g);
+
+		// Mapping status
+		g.gridx = 0;
+		g.gridy++;
+		JLabel l = new JLabel("By mapping status:");
+		l.setFont(headerFont);
+		add(l, g);
+
+		Map<CodeMapping.MappingStatus, Integer> mappingStats = new HashMap<>();
+		for (CodeMapping codeMapping : codeMappings) {
+			mappingStats.putIfAbsent(codeMapping.getMappingStatus(), 0);
+			Integer c = mappingStats.get(codeMapping.getMappingStatus());
+			mappingStats.put(codeMapping.getMappingStatus(), ++c);
+		}
+
+		mappingStats.forEach((key, value) -> {
+			g.gridx = 0;
+			g.gridy++;
+			add(new JLabel(String.format("%s - %d", key, value)), g);
+		});
+
+		// Equivalence status
+		g.gridx = 0;
+		g.gridy++;
+		l = new JLabel("By equivalence status:");
+		l.setFont(headerFont);
+		add(l, g);
+
+		Map<CodeMapping.Equivalence, Integer> equivalenceStats = new HashMap<>();
+		for (CodeMapping codeMapping : codeMappings) {
+			equivalenceStats.putIfAbsent(codeMapping.getEquivalence(), 0);
+			Integer c = equivalenceStats.get(codeMapping.getEquivalence());
+			equivalenceStats.put(codeMapping.getEquivalence(), ++c);
+		}
+
+		equivalenceStats.forEach((key, value) -> {
+			g.gridx = 0;
+			g.gridy++;
+			add(new JLabel(String.format("%s - %d", key, value)), g);
+		});
+
+		// Reviewer
+		g.gridx = 0;
+		g.gridy++;
+		l = new JLabel("By reviewer:");
+		l.setFont(headerFont);
+		add(l, g);
+
+		Map<String, Pair<Integer, Integer>> reviewerStats = new HashMap<>();
+		for (CodeMapping codeMapping : codeMappings) {
+			reviewerStats.putIfAbsent(codeMapping.getAssignedReviewer(), new Pair<>(0,0));
+			Pair<Integer, Integer> c = reviewerStats.get(codeMapping.getAssignedReviewer());
+			Integer nApproved = c.getItem1();
+			if (codeMapping.getMappingStatus() == CodeMapping.MappingStatus.APPROVED) {
+				c.setItem1(++nApproved);
+			}
+			Integer nTotal = c.getItem2();
+			c.setItem2(++nTotal);
```

Comment:
Same as above, you could use a `Stream.filter` to only count items where status is approved.